lib/datasource/artifactory/index.spec.ts

```diff
@@ -0,0 +1,146 @@
+import { getPkgReleases } from '..';
+import * as httpMock from '../../../test/http-mock';
+import { loadFixture } from '../../../test/util';
+import { EXTERNAL_HOST_ERROR } from '../../constants/error-messages';
+import { logger } from '../../logger';
+import { ArtifactoryDatasource } from '.';
+
+const datasource = ArtifactoryDatasource.id;
+
+const testRegistryUrl = 'https://jfrog.company.com/artifactory';
+const testLookupName = 'project';
+const testConfig = {
+  registryUrls: [testRegistryUrl],
+  depName: testLookupName,
+};
+
+function getPath(folder: string): string {
+  return '/' + folder;
+}
+
+describe('datasource/artifactory/index', () => {
+  beforeEach(() => {
+    jest.resetAllMocks();
+    jest.mock('../../logger');
+  });
+
+  describe('getReleases', () => {
+    it('parses real data (folders): with slash at the end', async () => {
+      httpMock
+        .scope(testRegistryUrl)
+        .get(getPath(testLookupName))
+        .reply(200, loadFixture('releases-as-folders.html'));
+      const res = await getPkgReleases({
+        ...testConfig,
+        datasource,
+        lookupName: testLookupName,
+      });
+      expect(res.releases).toHaveLength(4);
+      expect(res).toMatchSnapshot();
+    });
+
+    it('parses real data (files): without slash at the end', async () => {
+      httpMock
+        .scope(testRegistryUrl)
+        .get(getPath(testLookupName))
+        .reply(200, loadFixture('releases-as-files.html'));
+      const res = await getPkgReleases({
+        ...testConfig,
+        datasource,
+        lookupName: testLookupName,
+      });
+      expect(res.releases).toHaveLength(4);
+      expect(res).toMatchSnapshot();
+    });
+
+    it('parses real data (merge strategy with 2 registries)', async () => {
+      httpMock
+        .scope(testRegistryUrl)
+        .get(getPath(testLookupName))
+        .reply(200, loadFixture('releases-as-files.html'));
+      httpMock
+        .scope(testRegistryUrl)
+        .get(getPath(testLookupName))
+        .reply(200, '<html>\n<h1>Header</h1>\n<a>1.3.0</a>\n<hmtl/>');
+      const res = await getPkgReleases({
+        registryUrls: [testRegistryUrl, testRegistryUrl],
+        depName: testLookupName,
+        datasource,
+        lookupName: testLookupName,
+      });
+      expect(res.releases).toHaveLength(5);
+      expect(res).toMatchSnapshot();
```

Comment:
same

Reply:
For the merge strategy use case, it is not so simple to add the `testRegistryUrl`.
Instead, I have added a variable to distinguish the second registry which gives more understanding when checking the snapshot.